File: src/main/java/gr/jimmys/jimmysfoodzilla/repository/OrderRepository.java
package gr.jimmys.jimmysfoodzilla.repository;

import gr.jimmys.jimmysfoodzilla.dto.AllUserOrdersDTO;
import gr.jimmys.jimmysfoodzilla.models.Order;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface OrderRepository extends JpaRepository<Order, Integer> {
    @Query("SELECT new gr.jimmys.jimmysfoodzilla.dto.AllUserOrdersDTO(" +
            "o.totalCost, oi.id, o.id, d.id, oi.dishCounter, d.name, d.description, d.price) " +
            "FROM OrderItem oi " +
            "JOIN oi.order o " +
            "JOIN oi.dish d " +
            "WHERE o.user.id = :userId")
    List<AllUserOrdersDTO> findUserOrders(@Param("userId") String userId);
}

File: ReactMeals_WebApi/Repositories/OrderRepository.cs
using Microsoft.EntityFrameworkCore;
using ReactMeals_WebApi.Contexts;
using ReactMeals_WebApi.DTO;
using ReactMeals_WebApi.Models;

namespace ReactMeals_WebApi.Repositories;

public class OrderRepository(MainDbContext context)
{
    public async Task AddAsync(WebOrder order)
    {
        context.Orders.Add(order);
        await context.SaveChangesAsync();
    }
    public IQueryable<WebOrder> GetOrders()
    {
        return context.Orders;
    }

    public async Task<List<AllUserOrdersDTO>> GetUserOrdersAsync(string userId)
    {
        return await (from orderItem in context.OrderItems
                      join order in context.Orders on orderItem.WebOrderId equals order.Id
                      join dish in context.Dishes on orderItem.DishId equals dish.DishId
                      where order.UserId == userId
                      select new AllUserOrdersDTO(order.TotalCost, orderItem.Id, orderItem.WebOrderId,
                      orderItem.DishId, orderItem.Dish_counter,
                      dish.Dish_name, dish.Dish_description, dish.Price)).AsNoTracking().ToListAsync();
    }
}
